Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -322,6 +322,21 @@ TEST_F(SerializerTest, BadStringValue) {
       Status(FirestoreErrorCode::InvalidArgument, "ignored"), bytes);
 }
 
+TEST_F(SerializerTest, BadStringValue2) {
+  std::vector<uint8_t> bytes;
+  Status status =
+      serializer.EncodeFieldValue(FieldValue::StringValue("a"), &bytes);
+  EXPECT_TRUE(status.ok());
+
+  // Mutate that bytes: Claim that the string length is 5 instead of 1.
+  // (The first two bytes are used by the encoded tag.)
+  EXPECT_EQ(bytes[2], 1);
+  bytes[2] = 5;
```

Comment:
If we're doing this a bunch you might consider a helper like

```c++
void Mutate(uint8_t *byte, uint8_t expected_initial_value, uint8_t new_value) {
  ASSERT_EQ(*byte, expected_initial_value);
  *byte = new_value;
}
```
Alternatively a version that takes the vector and an offset, which would allow an overload that takes a series of bytes to replace if that's necessary.